ui/app/templates/components/secret-edit-display.hbs

```diff
@@ -9,6 +9,31 @@
     </div>
   {{/if}}
 
+  {{#if @showWriteWithoutReadWarning}}
+    {{#if (and @isV2 @model.failedServerRead)}}
+      <AlertBanner
+        @type="warning"
+        @message="We were not able to read metadata for this secret or the current version's data. Creating a new version of the secret with this form will not be able to use the check-and-set mechanism. If this is required on the secret, then you will need access to read the secret's metadata."
+        @class="is-marginless"
+        data-test-v2-no-cas-warning
+      />
+    {{else if @isV2}}
+      <AlertBanner
+        @type="warning"
+        @message="We were not able to read the current secret version to provide editing. Saving this form will create a new version of the secret and will utilize the available check-and-set mechanism."
```

Comment:
I like the way you specified on the previous pages that the policies are what prevent the user from being able to read secret metadata. I think "we were not able to read metadata" is a little ambiguous -- as a user, I might see that and think there was some kind of unexpected error (vs. expected). How about something like:

"Your policies prevent you from reading and editing the current secret version. Saving this form will create a new version of the secret and will utilize the available check-and-set mechanism."